src/harness/testcases/WINNF_FT_S_FAD_testcase.py

```diff
@@ -0,0 +1,189 @@
+            grant_ids.append(resp['grantId'])
+        # STEP 2
+        # Inject PPA
+        pal_low_frequency = 3550000000.0
+        pal_high_frequency = 3560000000.0
+        pal_record = json.load(
+          open(os.path.join('testcases', 'testdata', 'pal_record_0.json')))
+        ppa_record = json.load(
+          open(os.path.join('testcases', 'testdata', 'ppa_record_0.json')))
+        ppa_record, pal_record = makePpaAndPalRecordsConsistent(ppa_record,
+                                                                [pal_record],
+                                                                pal_low_frequency,
+                                                                pal_high_frequency,
+                                                                device_a['userId'])
+        self._sas_admin.InjectPalDatabaseRecord(pal_record[0])
+        self._sas_admin.InjectZoneData({'record': ppa_record})
+        # Inject Esc Sensor
+        esc_sensor_record = json.load(
+          open(os.path.join('testcases', 'testdata', 'esc_sensor_record_0.json')))
+        self._sas_admin.InjectEscSensorDataRecord({'record': esc_sensor_record})
+        # STEP 3
+        self._sas_admin.TriggerFullActivityDump()
+        # STEP 4
+        response = self._sas.GetFullActivityDump()
+        # STEP 5
+        self.assertContainsRequiredFields("FullActivityDump.schema.json", response)
+        # STEP 6 AND CHECK
+        self.assertGreaterEqual(3, len(response['files']))
+        self.assertLessEqual(2 + len(cbsd_ids), len(response['files']))
+        cbsd_dump_files = [dump_file for dump_file in response['files'] if dump_file['recordType'] ==  'cbsd']
+        esc_sensor_dump_file = [dump_file for dump_file in response['files'] if dump_file['recordType'] ==  'esc_sensor']
+        zone_dump_file = [dump_file for dump_file in response['files'] if dump_file['recordType'] ==  'zone']        
+        # Verify the record type
+        # Verify the response files with ActivityDumpFile.schema.json Object schema
+        for dump_file in [esc_sensor_dump_file, zone_dump_file]:
+            self.assertContainsRequiredFields("ActivityDumpFile.schema.json",
+                                               dump_file[0])
+            self.assertEqual(1, len(dump_file))
+        self.assertGreaterEqual(1, len(cbsd_dump_files))
+        self.assertLessEqual(len(cbsd_dump_files), len(cbsd_ids))
+        data = []
+        for dump_file in cbsd_dump_files:
+            self.assertContainsRequiredFields("ActivityDumpFile.schema.json",
+                                               dump_file)
+            data.append(self._sas.DownloadFile(dump_file['url']))
+        # Get json data from url
+        #for dump_file in cbsd_dump_files    
+        # Verify that everything in the full dump matches a cbsd or grant created at the beginning
+        cbsd_records = [cbsd_record['recordData'] for cbsd_record in data]
+        for record in cbsd_records:
```

Comment:
Consider adding a comment that duplicate records will be detected in the assertCbsdRecord() function.